lib/rpush/daemon/store/active_record.rb

```diff
@@ -29,13 +28,21 @@ def all_apps
 
         def deliverable_notifications(limit)
           with_database_reconnect_and_retry do
-            Rpush::Client::ActiveRecord::Notification.transaction do
+            notifications = Rpush::Client::ActiveRecord::Notification.transaction do
               relation = ready_for_delivery
               relation = relation.limit(limit)
-              notifications = claim(relation)
-              mark_processing(notifications)
-              notifications
+              ids = relation.lock(true).pluck(:id)
```

Comment:
Can you use `.ids` instead of `.pluck(:id)` here? Should essentially be the same.